src/audioin.js

```diff
@@ -224,25 +225,43 @@ define(function (require) {
     }
   };
 
-  /**
-   *  Returns a list of available input sources. Some browsers
-   *  give the client the option to set their own media source.
-   *  Others allow JavaScript to determine which source,
-   *  and for this we have listSources() and setSource().<br/>
-   *
-   *  @method  listSources
-   *  @return {Array}
-   */
   p5.AudioIn.prototype.listSources = function() {
+    console.log('listSources is deprecated - please use AudioIn.getSources')
     console.log('input sources: ');
     console.log(p5sound.inputSources);
     if (p5sound.inputSources.length > 0) {
-      return p5sound.inputSources;
+      //return p5sound.inputSources;
     } else {
       return 'This browser does not support MediaStreamTrack.getSources()';
     }
   };
-
+  /**
+   * Returns a list of available input sources. Chrome allows the 
+   * the user to access a list of possible media sources. Firefox 
+   * allows the user to choose from input sources in the permissions
+   * dialogue.
+   *
+   * @method  getSources
+   * @param  {Function} callback [a callback to handle the sources 
+   * when they have been enumerated]
+   * @return {[type]}            [description]
+   */
```

Comment:
awesome, thanks @polyrhythmatic! You could add an in-line example to the documentatino following [this guide](https://github.com/processing/p5.js/wiki/Inline-documentation#adding-example-code), and also put a sketch that you were using for testing into the /examples folder of this repo.

Have you had a chance to try this out in the Desktop Editor?